### PR TITLE
nrf: Add NVMC peripheral (microbitfs) support.

### PR DESCRIPTION
**Pull request by aykevl**
There is no support yet for a SoftDevice.

It also fixes a potentially serious bug in start_index generation.

### CODE REVIEW

**Comment by glennrub** [comment]
Great, Ayke!

Why not add the nrf_nvmc.c to SRC_NRFX?

**Comment by glennrub** [comment]
Second thought, they have different paths. Maybe we should change change SRC_NRFX to SRC_NRFX_DRIVERS. Hence, we can add SRC_NRF_HAL.

File: ports/nrf/modules/uos/microbitfs.c [hunk 1]
```diff
@@ -31,8 +31,8 @@
 #include <sys/stat.h>
 
 #include "microbitfs.h"
-#include "hal/hal_nvmc.h"
-#include "hal/hal_rng.h"
+#include "nrf_nvmc.h"
+#include "modrandom.h"
```
Comment (glennrub) [inline ports/nrf/modules/uos/microbitfs.c]: Did you forget to commit this header as part of this patch? :)

Comment (aykevl) [reply ports/nrf/modules/uos/microbitfs.c]: You mean modrandom?
It is intended to be there. On startup, a random start index is chosen, supposedly for wear levelling.

I have started doubting whether that is truly necessary, due to the way the filesystem works: it fully writes the complete flash area and only when it is full it does a kind of garbage collection to erase freed blocks. The complete flash area is erased on such a cycle, so a random start index makes little or no difference.

Comment (aykevl) [reply ports/nrf/modules/uos/microbitfs.c]: Oh, sorry, I misread your comment. Yes, it looks like I forgot it.

Comment (aykevl) [reply ports/nrf/modules/uos/microbitfs.c]: Studying the FS a bit more shows the random start index is indeed necessary.

**Pull request merged**